googledirectory/table_googledirectory_group.go

```diff
@@ -109,22 +116,54 @@ func listDirectoryGroups(ctx context.Context, d *plugin.QueryData, _ *plugin.Hyd
 		return nil, err
 	}
 
+	var queryFilter, query string
+	var filter []string
+
+	if d.KeyColumnQuals["name"] != nil {
+		filter = append(filter, fmt.Sprintf("name='%s'", d.KeyColumnQuals["name"].GetStringValue()))
+	}
+
+	if d.KeyColumnQuals["query"] != nil {
+		queryFilter = d.KeyColumnQuals["query"].GetStringValue()
+	}
+
+	if queryFilter != "" {
+		query = queryFilter
+	} else if len(filter) > 0 {
+		query = strings.Join(filter, " ")
+	}
+
+	// Since, query parameter can't be empty, set default param name:**, to return all groups
+	if query == "" {
+		query = "name:**"
+	}
+
 	// Set default value to my_customer, to represent current account
 	customerID := "my_customer"
 	if d.KeyColumnQuals["customer_id"] != nil {
 		customerID = d.KeyColumnQuals["customer_id"].GetStringValue()
 	}
 
-	query := d.KeyColumnQuals["query"].GetStringValue()
-	// Since, query parameter can't be empty, set default param name:**, to return all groups
-	if query == "" {
-		query = "name:**"
+	// By default, API can return maximum 200 records in a single page
```

Comment:
What effect does using a larger page size vs smaller page size have on the SDK call, the Google Admin API, or anything else, i.e., response time, throttling, optimizations with context cancelling?

Does Google have any best practices on what paging size to use based on use case?

Reply:
After talking with @Subhajit97 , we've decided on going with the larger page sizes, as we'd like to reduce the number of requests when possible due to throttling and hourly/daily limits on API requests.